src/game/editor/editor2.cpp

```diff
@@ -2405,6 +2413,30 @@ bool CEditor2::SaveFileCallback(const char *pFilepath, void *pContext)
 	return pEditor->SaveMap(pFilepath);
 }
 
+void CEditor2::InvokeNewCb(bool Choice, void *pContext)
+{
+	CEditor2 *pEditor = (CEditor2 *)pContext;
+	pEditor->ExitPopup();
+
+	if(!Choice)
+		return;
+
+	pEditor->SaveMap(pEditor->m_aCurrentMapPath);
+	pEditor->Reset();
+}
+
+void CEditor2::InvokeLoadPopupCb(bool Choice, void *pContext)
+{
+	CEditor2 *pEditor = (CEditor2 *)pContext;
+	pEditor->ExitPopup();
+
+	if(!Choice)
+		return;
+
+	pEditor->SaveMap(pEditor->m_aCurrentMapPath);
```

Comment:
This does not work when the current path is empty, a 'Save As' popup should be pushed then. But that complicates things a bit, so perhaps if we do push the Save As popup, let the user do the File > Load again (at least for now).
Best way would be to have a single function that pushes the Save As popup when we try to simply Save with an empty name (or disallowed name like default maps?), since File > Save should have the same behaviour.